src/lib/network_pool/intf.ml

```diff
@@ -0,0 +1,113 @@
+open Async_kernel
+open Core_kernel
+open Coda_base
+open Pipe_lib
+
+module type Resource_pool_base_intf = sig
```

Comment:
Personal nit but if I have module types defined in intf file, I usually omit the `_intf` suffix. If it's in an intf file, it is already implied that the module type is an interface. It feels redundant.

Reply:
Most of our code always uses a `_intf` suffix, so I will stick to that style for now. Granted, I'm not a fan of unecessary the `_intf`s either, it's just the standard we are using.